File: largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.py
class Solution(object):
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Solution: O(n) time, O(1) additional space.
        
        max_val = max(nums)
        max_idx = nums.index(max_val)
        
        for num in nums:
            if num != max_val and num * 2 > max_val:
                return -1
            
        return max_idx

File: largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.java
class Solution {
    
    public int dominantIndex(int[] nums) {
        
        // Solution: O(n) time, O(1) additional space.
        
        int maxIdx = 0;
        for (int i = 1; i < nums.length; i++) maxIdx = nums[i] > nums[maxIdx] ? i : maxIdx;
        int maxVal = nums[maxIdx];
        
        for (int num: nums) if (num != maxVal && num * 2 > maxVal) return -1;
            
        return maxIdx;
        
    }
    
}